etl/pypasar/omop/condition_occurrence.py

```diff
@@ -22,15 +28,86 @@ def execute(self):
             raise err
 
     def initialize(self):
-        pass
+        # Truncate
+        with self.engine.connect() as connection:
+            with connection.begin():
+                connection.execute(text(f"Truncate table {self.omop_schema}.condition_occurrence"))
 
     def process(self):
-        # In batches
-        # Read from source
-        # Transform
-        # Ingest into OMOP Table
-        pass
+        total_count_source_postop_discharge = self.fetch_total_count_source_postop_discharge()
+        print(f"Total count {total_count_source_postop_discharge}")
+        while self.offset <= total_count_source_postop_discharge: # Fetch and process in batches
+            source_batch = self.retrieve()
+            transformed_batch = self.transform(source_batch)
+            del source_batch
+            self.ingest(transformed_batch)
+            del transformed_batch
+            gc.collect()
+            self.offset = self.offset + self.limit
+
+    def retrieve(self):
+        source_batch = self.fetch_in_batch_source_postop_discharge()
+        source_postop_discharge_df = pd.DataFrame(source_batch.fetchall())
+        source_postop_discharge_df.columns = {'anon_case_no': str, 'id': int, 'diagnosis_date': 'datetime64[ns]', 'diagnosis_code': str, 'session_id': int}.keys()
+        print(source_postop_discharge_df.head(1))
+        print(f"offset {self.offset} limit {self.limit} batch_count {len(source_postop_discharge_df)} retrieved..")
+        return source_postop_discharge_df
+    
+    def transform(self, source_batch):
+        condition_occurrence_schema = {
+            'condition_occurrence_id': int,
+            'person_id': int,
+            'condition_concept_id': int,
+            'condition_start_date': 'datetime64[ns]',
+            'condition_start_datetime': 'datetime64[ns]',
+            'condition_end_date': 'datetime64[ns]',
+            'condition_end_datetime': 'datetime64[ns]',
+            'condition_type_concept_id': int,
+            'condition_status_concept_id': int,
+            'visit_occurrence_id': int,
+            'condition_source_value': str
+        }
+        # Initialize dataframe and display columns info
+        condition_occ_df = pd.DataFrame(columns=condition_occurrence_schema.keys()).astype(condition_occurrence_schema)
+        # print(f"source {len(source_batch)}")
+        
+        if len(source_batch) > 0:
+            condition_occ_df['condition_start_date'] = pd.to_datetime(source_batch['diagnosis_date']).dt.date
+            condition_occ_df['condition_start_datetime'] = source_batch['diagnosis_date']
+            condition_occ_df['condition_end_date'] = pd.to_datetime(source_batch['diagnosis_date']).dt.date
```

Comment:
Since `condition_end_date` and `condition_end_datetime` lack source mappings and are optional, consider setting them to NULL

Reply:
Updated